README.md

```diff
@@ -2,9 +2,19 @@
 
 A tool for GUI automation using a variety of computer vision and desktop control backends.
 
+## Introduction
+
+In order to do GUI automation you usually need to solve two problems: first, you need to have a way to control and interact with the interface and platform you are automating and second, you need to be able to locate the objects you are interested in on the screen. Guibot helps you do both.
+
+To interact with GUIs, guibot provides the [`desktopcontrol`](https://github.com/intra2net/guibot/blob/master/guibot/desktopcontrol.py) module which contains a common interface for different desktop control backends, with methods to move the mouse, take screenshots, type characters and so on. The backend to use will depend on how your interface is accessible, but the ones currently supported are qemu, vnctotool, xdotool, and autopy.
+
+To locate an element on the screen, you will need an image representing the screen, a [`target`](https://github.com/intra2net/guibot/blob/master/guibot/target.py) and a [`finder`](https://github.com/intra2net/guibot/blob/master/guibot/finder.py) - the finder looks for the target on the image, and returns the coordinates to the region where that target appears. Finders, like desktop controllers, are wrappers around different backends that guibot supports (OpenCV, TesseractOCR, PyTorch, and autopy), while targets are a representation of the object you are looking for, such as images, text, or patterns.
+
+Finally, to bridge the gap between controlling the GUI and finding elements, the [`region`](https://github.com/intra2net/guibot/blob/master/guibot/region.py) module is provided. It represents a subregion of a screen, and contains methods to locate targets in this region and interact with the graphical interface using the chosen backends.
```

Comment:
I will double check this once more once we decide some of the above for things like "destkop controllers".

Reply:
This sounds better now, perhaps listing the backends is too much considering we already list them at the end of the README in "Supported backends". Instead in the brackets we could just mention `(e.g. OpenCV)` or `(for extensive list see below)` or `(see below)`. Otherwise I like the three paragraphs and the way the relate the left, right, and center.